block-node/stream-publisher/src/main/java/org/hiero/block/node/stream/publisher/PublisherHandler.java

```diff
@@ -600,6 +615,7 @@ private BatchHandleResult handleEndBehind() {
      * Handle the END_DUPLICATE action for a block.
      */
     private BatchHandleResult handleEndDuplicate() {
+        LOGGER.log(DEBUG, "Handler {0} is sending DUPLICATE_BLOCK", handlerId);
```

Comment:
Same here, include the last known persisted block number:
```suggestion
        LOGGER.log(DEBUG, "Handler {0} is sending DUPLICATE_BLOCK({1}).", handlerId, publisherManager.getLatestBlockNumber());
```